### PR TITLE
Udate upload-artifact and download-artifact Github Actions

### PR DESCRIPTION
**Pull request by esseivaju**
This was probably missed during an update, as everywhere else already uses `upload-artifact@v4`. [V3 will be deprecated soon](https://github.blog/changelog/2024-11-05-notice-of-breaking-changes-for-github-actions/).

### CODE REVIEW

**Comment by sethrj** [comment]
@esseivaju ~~it doesn't work on the docker images because their OS is too old for the Node.js version.~~ I guess v4 didn't work on ROCm in particular and now I've had to disable it. As I said I was going to update those images this week and the docker CI script that corresponds to it.

File: .github/workflows/pull_request_completed.yml [hunk 1]
```diff
@@ -18,11 +18,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Download and Extract Artifacts
-      uses: dawidd6/action-download-artifact@e7466d1a7587ed14867642c2ca74b5bcc1e19a2d # v6
+      uses: actions/download-artifact@v4
```
Comment (pcanal) [inline .github/workflows/pull_request_completed.yml:21]: The 2 are not quite the same. The former allows to `download this artifact in another workflow that is run after the first one.`.  Do we need that feature?

Comment (sethrj) [reply .github/workflows/pull_request_completed.yml:21]: Yes we do.

Comment (esseivaju) [reply .github/workflows/pull_request_completed.yml:21]: `download-artifact@v4`  should now [support](https://github.com/actions/download-artifact/tree/v4?tab=readme-ov-file#download-artifacts-from-other-workflow-runs-or-repositories) this as well

**Comment by esseivaju** [comment]
> @esseivaju ~it doesn't work on the docker images because their OS is too old for the Node.js version.~ I guess v4 didn't work on ROCm in particular and now I've had to disable it. As I said I was going to update those images this week and the docker CI script that corresponds to it.

Sure, I guess you can update everything in one go with the new images then.

File: .github/workflows/pull_request_completed.yml (hunk 1 above)
Comment (sethrj) [reply .github/workflows/pull_request_completed.yml:21]: ooh good to know!

**Comment by sethrj** [comment]
By the way @esseivaju , all changes to `pull_request_completed` have to be tested on `develop` on a personal repo before deploying to celeritas: those changes won't be tested as part of the PR (since it's the PR action on the *develop* branch that gets triggered), so I'd be happy to have those separate from the CI update.

**Comment by esseivaju** [comment]
@sethrj Sounds good; I'll test the update to `pull_request_completed` and re-open a new PR later.

**Pull request closed without merging**